group_project_v2/stage/base.py

```diff
@@ -325,6 +326,8 @@ def navigation_view(self, context):
         return fragment
 
     def dashboard_view(self, context):
+
+        self.check_dashboard_access(self.user_id)
```

Comment:
This make sense, but this would interact with ``user_has_no_access_to_protected_view`` (or howewer it will be renamed). So if ``user_has_no_access_to_protected_view`` decorator would just raise an exception then: 

     @user_has_no_access_to_protected_view
     @check_dashboard_access
     def foo(args):
         pass

would behave differently than: 

     @check_dashboard_access
     @user_has_no_access_to_protected_view
     def foo(args):
         pass

This is acceptable in Python, but has downsides.